ch_integration_tests/tests/integration.rs

```diff
@@ -67,56 +67,87 @@ mod tests {
 
     impl<'a> Guest<'a> {
         fn create_domain(&self, vcpus: VcpuConfig, memory_size: u64) -> PathBuf {
-            let domain = format!(
-                "<domain type='ch'> \
-        <name>{}</name> \
-        <uuid>{}</uuid> \
-        <genid>43dc0cf8-809b-4adb-9bea-a9abb5f3d90e</genid> \
-        <title>Test VM {}</title> \
-        <description>Test VM {}</description> \
-        <os> \
-                <type>hvm</type> \
-                <kernel>{}</kernel> \
-        </os> \
-        <vcpu current='{}'>{}</vcpu> \
-        <memory unit='b'>{}</memory> \
-        <devices> \
-                <disk type='file'> \
-                        <source file='{}'/> \
-                        <target dev='vda' bus='virtio'/> \
-                </disk> \
-                <disk type='file'> \
-                        <source file='{}'/> \
-                        <target dev='vdb' bus='virtio'/> \
-                </disk> \
-                <console type='pty'> \
-                        <target type='virtio' port='0'/> \
-                </console> \
-                <interface type='ethernet'> \
-                        <mac address='{}'/> \
-                        <model type='virtio'/> \
-                        <source> \
-                               <ip address='{}' prefix='24'/> \
-                        </source> \
-              </interface> \
-        </devices> \
-        </domain>",
-                self.vm_name,
-                self.uuid,
-                self.vm_name,
-                self.vm_name,
-                self.kernel_path.to_str().unwrap(),
-                vcpus.boot,
-                vcpus.max,
-                memory_size,
-                self.disk_config
-                    .disk(DiskType::OperatingSystem)
-                    .unwrap()
-                    .as_str(),
-                self.disk_config.disk(DiskType::CloudInit).unwrap().as_str(),
-                self.network.guest_mac,
-                self.network.host_ip,
-            );
+            let xml_domain = XMLElement::new("domain")
```

Comment:
You not thinking that you could use the sister https://crates.io/crates/simple_xml_serialize_macro crate to do all this automatically from a struct?

Reply:
I started with the macro approach, but realized it does not handle `Option<>`, which I will need to add `cmdline` only when needed with direct kernel boot.

Reply:
Urgh.